datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -632,22 +631,21 @@ impl LogicalPlan {
     /// params_values
     pub fn replace_params_with_values(
         &self,
-        param_values: &Vec<ScalarValue>,
+        param_values: &[ScalarValue],
     ) -> Result<LogicalPlan, DataFusionError> {
```

Comment:
Perhaps its good to reuse existing `Result<LogicalPlan>` which already `Result<T, DataFusionError>`. It will make the code less wordy

Reply:
Thank you @comphead. Actually, I have noticed this, but as we always use `Result<_, DatafusionError>` in the file `plan.rs`, I didn't change it. 

It is a good idea to file an issue to clean the `Result<_, DatafusionError>` usages in the project. But I think a more elegant way it to try to implement our own lint so that we can find it during the `cargo clippy`.

I've searched how to implement custom lints, but made less progress.

Reply:
Sounds good, I'll create a separate ticket to clean up. Lints are great ideas, not sure how easy to implement the custom ones

Reply:
Thank you @comphead. I will also file an issue to track the `custom lint` and find if there are more needs in our project to enable it.